test/e2e/set/scyllacluster/scyllamanager.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2021 ScyllaDB
+
+package scyllacluster
+
+import (
+	"context"
+	"encoding/json"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	scyllav1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1"
+	scyllaclusterfixture "github.com/scylladb/scylla-operator/test/e2e/fixture/scyllacluster"
+	"github.com/scylladb/scylla-operator/test/e2e/framework"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = g.Describe("Scylla Manager integration", func() {
+	defer g.GinkgoRecover()
+
+	f := framework.NewFramework("scyllacluster")
+
+	g.It("should discover cluster and sync tasks", func() {
+		ctx, cancel := context.WithTimeout(context.Background(), testTimout)
+		defer cancel()
+
+		sc := scyllaclusterfixture.BasicScyllaCluster.ReadOrFail()
+		sc.Spec.Datacenter.Racks[0].Members = 1
+		sc.Spec.Repairs = append(sc.Spec.Repairs, scyllav1.RepairTaskSpec{
+			SchedulerTaskSpec: scyllav1.SchedulerTaskSpec{
+				Name:     "weekly repair",
+				Interval: "7d",
+			},
+			Parallel: 123,
+		})
+
+		// Backup scheduling is going to fail because location is not accessible in our env.
+		sc.Spec.Backups = append(sc.Spec.Backups, scyllav1.BackupTaskSpec{
+			SchedulerTaskSpec: scyllav1.SchedulerTaskSpec{
+				Name:     "daily backup",
+				Interval: "1d",
+			},
+			Location: []string{"s3:bucket"},
+		})
+
+		framework.By("Creating a ScyllaCluster")
+		err := framework.SetupScyllaClusterSA(ctx, f.KubeClient().CoreV1(), f.KubeClient().RbacV1(), f.Namespace(), sc.Name)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		sc, err = f.ScyllaClient().ScyllaV1().ScyllaClusters(f.Namespace()).Create(ctx, sc, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for the ScyllaCluster to deploy")
+		waitCtx1, waitCtx1Cancel := contextForRollout(ctx, sc)
+		defer waitCtx1Cancel()
+		sc, err = waitForScyllaClusterState(waitCtx1, f.ScyllaClient().ScyllaV1(), sc.Namespace, sc.Name, scyllaClusterRolledOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		verifyScyllaCluster(ctx, f.KubeClient(), sc)
+
+		framework.By("Waiting for the cluster sync with Scylla Manager")
+		registeredInManagerCond := func(sc *scyllav1.ScyllaCluster) (bool, error) {
+			return sc.Status.ManagerID != nil, nil
+		}
+
+		repairTaskScheduledCond := func(name string) func(sc *scyllav1.ScyllaCluster) (bool, error) {
```

Comment:
This is always using the 1 fixed name so you don't really need the closure here if you stop shadowing `sc` or if you initialize the name from a variable at the beginning. Alternatively you can move the function out of the test to be reusable.